swiftide-integrations/src/kafka/persist.rs

```diff
@@ -0,0 +1,110 @@
+use std::{sync::Arc, time::Duration};
+
+use anyhow::Result;
+use async_trait::async_trait;
+
+use rdkafka::producer::FutureRecord;
+use swiftide_core::{
+    Persist,
+    indexing::{IndexingStream, Node},
+};
+
+use super::Kafka;
+
+#[async_trait]
+impl Persist for Kafka {
+    async fn setup(&self) -> Result<()> {
+        self.create_topic_if_not_exists().await?;
+        Ok(())
+    }
+
+    fn batch_size(&self) -> Option<usize> {
+        None
+    }
```

Comment:
Since you also implemented batch_size, why does this return null? The pipeline will only persist in batch if the batch size is set https://github.com/bosun-ai/swiftide/blob/master/swiftide-indexing/src/pipeline.rs#L344